config/crd/bases/infrastructure.cluster.x-k8s.io_openstackclusters.yaml

```diff
@@ -5269,11 +5269,28 @@ spec:
                             type: string
                         type: object
                       serverMetadata:
-                        additionalProperties:
-                          type: string
                         description: Metadata mapping. Allows you to create a map
                           of key value pairs to add to the server instance.
-                        type: object
+                        items:
+                          properties:
+                            key:
+                              description: |-
+                                Key is the server metadata key
+                                kubebuilder:validation:MaxLength:=255
+                              type: string
+                            value:
+                              description: |-
+                                Value is the server metadata value
+                                kubebuilder:validation:MaxLength:=255
+                              type: string
+                          required:
+                          - key
+                          - value
+                          type: object
+                        type: array
+                        x-kubernetes-list-map-keys:
+                        - key
+                        x-kubernetes-list-type: map
```

Comment:
Will this have more consequences, like e.g. not allowing some characters because K8s doesn't allow them, but OpenStack does?

Reply:
I don't think there are any constraints on map keys. [The documentation](https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy) only says:
> The key fields must be scalars.
They don't have the same constraints as names.